### ISSUE TITLE
Lost all items when Luggage was deleted by Mob Slaughter Factory

### ISSUE THREAD

**Issue by tehbiga** [GizmoTheMoonPig/luggage#20]
### Forge or Fabric

Forge

### Mod Version

1.18.2

### Bug Description

**tl;dr:**
Luggage decided to walk into a Mob Slaughter Factory for the 8 ticks I had opened it to swap a block out and it just poofed out of existance. No items were produced, they just were all gone.

**Life Story:**
Not sure if this is within your power to prevent, but I've had a Luggage with all my tools since for whatever reason I didn't have a backpack yet.

### Steps to Reproduce

1. Create Mob Slaughter Factory
2. Power factory
3. Lure Luggage into it's working area

### Please Read and Confirm The Following

- [X] I have confirmed the details provided in this report are concise as possible and does not contained vague information (ie. Versions are properly recorded, answers to questions are clear).
- [X] I have confirmed this issue is unique and has not been reported already.

**Comment by GizmoTheMoonPig** [comment]
I'll look into how the mob slaughter factory works. That really shouldn't be happening. If it's something I can't fix I'll let them know over there

**Comment by GizmoTheMoonPig** [comment]
it looks like they check if the entity is alive, and if it is they remove it. They really should be checking if the entity can be hurt before doing this honestly. I'll let them know and link the issue back to here. Thanks for reporting!

**Comment by GizmoTheMoonPig** [comment]
I made luggage drop all items if it dies, which should stop you from losing things. The part about them dying isn't fixable on my end, you'll have to wait for Industrial Foregoing to do something about that part (I opened an issue for it here when you reported this: https://github.com/InnovativeOnlineIndustries/Industrial-Foregoing/issues/1308). Appreciate the report, and I'm sorry you lost your items. Hope it doesn't happen again :(
